app/assets/javascripts/map.js

```diff
@@ -0,0 +1,44 @@
+var map;
+$(document).ready(function(){
+  map = new GMaps({
+    div: '#map',
+    lat: 57.7089,
+    lng: 11.9746
+  });
+  performGeolocation();
+  addMarker();
+});
+function addMarker() {
+  for (i = 0; i < gon.restaurants.length; i++) {
+    map.addMarker({
+        lat: gon.restaurants[0].latitude,
+        lng: gon.restaurants[0].longitude,
+    });
+  }
+}
+
+function performGeolocation() {
+  var latitude;
+  var longitude;
+  var testing_env = $('#map').data('test-env');
+  if(testing_env == false) {
+    GMaps.geolocate({
+              success: function (position) {
+                latitude = position.coords.latitude
+                longitude = position.coords.longitude
+                map.setCenter(latitude, longitude);
```

Comment:
This can be performed as the last thing in this function. We repeat the same command in the else block.